### ISSUE TITLE
cast to map with clickhouse dialect ignores case

### ISSUE THREAD

**Issue by cpcloud** [tobymao/sqlglot#551]
Yet another clickhouse case sensitivity bug :)

ClickHouse CLI:

```
albatross :) select cast(map('a', 1, 'b', 2) as Map(String, Int8));

SELECT CAST(map('a', 1, 'b', 2), 'Map(String, Int8)')

Query id: bab188fb-69f7-4c0f-9bde-ac524a08f6e8

┌─CAST(map('a', 1, 'b', 2), 'Map(String, Int8)')─┐
│ {'a':1,'b':2}                                  │
└────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```

sqlglot:

```
In [7]: sg.parse_one("cast(map('a', 1, 'b', 2) as Map(String, Int8))", read="clickhouse").sql(dialect="clickhouse")
Out[7]: "CAST(map('a', 1, 'b', 2) AS MAP(TEXT, TINYINT))"
```

Perhaps there's a way to disable all case transformations for the clickhouse dialect.

**Comment by cpcloud** [comment]
Seems to be happening with other types as well:

```
In [2]: sg.parse_one("cast([] as Array(Nullable(Int8)))", read="clickhouse").sql(dialect="clickhouse")
Out[2]: 'CAST([] AS ARRAY(Nullable(TINYINT)))'
```

**Comment by tobymao** [comment]
fixed by #552
